client/signup/controller.js

```diff
@@ -63,6 +63,14 @@ const removeWhiteBackground = function () {
 	document.body.classList.remove( 'is-white-signup' );
 };
 
+const gutenbergRedirect = function () {
+	if ( 'verticals' === abtest( 'verticalsOnGutenboarding' ) ) {
+		window.location.replace( window.location.origin + '/new?vertical' + window.location.search );
```

Comment:
I noticed a small bug with this that would be good to fix while we're at it but okay for follow-up, too:

That `location.search` starts with `?` so we end with `/new?vertical&?something`

Something like this could work:

```js
`${window.location.origin}/new${(window.location.search ? window.location.search + '&vertical' : '?vertical')}`
```

...or at this point might be easiest to use `new URL()` or `new URLSearchParams()` or some query args `/lib/*` we probably have around in Calypso.

Reply:
Your suggestion works.
![image](https://user-images.githubusercontent.com/1287077/89999116-aa5e6080-dc8e-11ea-95a1-9fcd8c49f694.png)

Reply:
Actually why to deal with all this `window.location.replace` stuff if we could just:

```js
const url = new URL( window.location );
url.pathname = '/new';

if ( 'verticals' === abtest( 'verticalsOnGutenboarding' ) ) {
  url.searchParams.append( 'vertical', '' );
}

window.location.replace( url.toString() );
```

...it'll save everything the URL previously had, including hashes and query args, and is very easy to read and expand later.

Reply:
Okay with the latest suggestion this is how it looks like. There's a `=` at the end but it works. :)

![image](https://user-images.githubusercontent.com/1287077/89999693-6e77cb00-dc8f-11ea-8cbe-c7c5c92028eb.png)

Reply:
We'll anyway remove the URL arg once the abtest for verticals is over but we got rid of pesky `window.location.search` which is nice :-)